examples/python/reconstruction_system/make_fragments.py

```diff
@@ -172,13 +173,14 @@ def run(config):
         math.ceil(float(n_files) / config['n_frames_per_fragment']))
 
     if config["python_multi_threading"] is True:
-        from joblib import Parallel, delayed
-        import multiprocessing
-        import subprocess
-        MAX_THREAD = min(multiprocessing.cpu_count(), n_fragments)
-        Parallel(n_jobs=MAX_THREAD)(delayed(process_single_fragment)(
-            fragment_id, color_files, depth_files, n_files, n_fragments, config)
-                                    for fragment_id in range(n_fragments))
+        max_workers = min(max(1, multiprocessing.cpu_count() - 1), n_fragments)
+        # Prevent over allocation of open mp threads in child processes
+        os.environ['OMP_NUM_THREADS'] = '1'
+        mp_context = multiprocessing.get_context('spawn')
+        with mp_context.Pool(processes=max_workers) as pool:
+            args = [(fragment_id, color_files, depth_files, n_files,
+                     n_fragments, config) for fragment_id in range(n_fragments)]
+            pool.starmap(process_single_fragment, args)
```

Comment:
Thanks! Can you check timing with and without OMP_NUM_THREADS=1? 
Also, there are 2 other places that use `joblib` (`register_fragments` and `refine_registration`) Can you update those 2 files too?

Reply:
@ssheorey - Without OMP_NUM_THREADS=1 on full dataset, the code was running for 2 hours (almost half fragments finished) after which I stopped it. If not wrong, at this moment we are measuring time spent in context switching of threads rather than instruction execution time. I tried letting it run over night, but woke up with Pycharm and multiple chrome tabs killed by OS.

For subset of dataset - see `python_mp_forkserver`, `python_mp_spawn` in above comments. They are result without setting OMP_NUM_THREADS.

```
# On 16 core machine

# Multiprocessing - With OMP_NUM_THREADS=1 and 15 processes
- Making fragments    0:09:09.990264
- Register fragments  0:02:19.245263
- Refine registration 0:00:40.425758

# Multiprocessing - OMP_NUM_THREADS not set. 15 processes
- Making fragments    I will update this part later. Will let it run over night.
- Register fragments  0:02:52.160323
- Refine registration 0:01:33.050342

# Serial Execution 
- Making fragments    1:11:38.828078
- Register fragments  0:07:00.667959
- Refine registration 0:03:40.475970
```

Reply:
@ssheorey I have added all profiling results.

Reply:
Thanks! looks good.